examples_apps/run_minimizing_correlations.py

```diff
@@ -0,0 +1,355 @@
+phi[10:] = 0.
+phi /= np.linalg.norm(phi)
+
+
+######################################################################
+# Next, we define the creation operator :math:`\hat{a}`,
+#
+
+a = np.diag(np.sqrt(np.arange(1, cutoff)), k=1)
+
+
+######################################################################
+# the number operator :math:`\hat{n}=\hat{a}^\dagger \hat{a}`,
+#
+
+n_opt = a.T @ a
+
+
+######################################################################
+# and the quadrature operators :math:`\hat{x}=a+a^\dagger`,
+# :math:`\hat{p}=-i(a-a^\dagger)`.
+#
+
+# Define quadrature operators
+x = a + a.T
+p = -1j*(a-a.T)
+
+
+######################################################################
+# We can now calculate the displacement of the states in the phase space,
+# :math:`\alpha=\langle \psi \mid\hat{a}\mid\psi\rangle`. The following
+# function calculates this displacement, and then displaces the state by
+# :math:`-\alpha` to ensure it has zero displacement.
+#
+
+def recenter(state):
+    alpha = state.conj() @ a @ state
+    disp_alpha = expm(alpha.conj()*a - alpha*a.T)
+    out_state = disp_alpha @ state
+    return out_state
+
+
+######################################################################
+# First, let’s have a look at the displacement of state :math:`\ket{\psi}`
+# and state :math:`\ket{\phi}`:
+#
+
+print(psi.conj().T @ a @ psi)
+print(phi.conj().T @ a @ phi)
+
+
+######################################################################
+# Now, let’s center them in the phase space:
+#
+
+psi = recenter(psi)
+phi = recenter(phi)
+
+
+######################################################################
+# Checking they now have zero displacement:
+#
+
+print(np.round(psi.conj().T @ a @ psi, 9))
+print(np.round(phi.conj().T @ a @ phi, 9))
+
+
+######################################################################
+# Performing the optimization
+# ---------------------------
+#
+# We can construct the variational quantum circuit cost function, using Strawberry
+# Fields:
+#
+
+penalty_strength = 10
+
+prog = sf.Program(2)
+eng = sf.Engine("tf", backend_options={"cutoff_dim": cutoff})
+
+psi = tf.cast(psi, tf.complex64)
+phi_var_re = tf.Variable(phi.real)
+phi_var_im = tf.Variable(phi.imag)
+
+
+with prog.context as q:
+    Ket(prog.params("input_state")) | q
+    BSgate(np.pi/4, 0) | q
+
+
+
+######################################################################
+# Here, we are initializing a TensorFlow variable ``phi_var`` representing
+# the initial state of mode ``q[1]``, which we will optimize over. Note
+# that we take the outer product
+# :math:`\ket{in}=\ket{\psi}\otimes\ket{\phi}`, and use the ``Ket``
+# operator to initialise the circuit in this initial multimode pure state.
+
+
+def cost(phi_var_re, phi_var_im):
+    phi_var = tf.cast(phi_var_re, tf.complex64) + 1j*tf.cast(phi_var_im, tf.complex64)
+
+    in_state = tf.einsum('i,j->ij', psi, phi_var)
```

Comment:
Just a note that `einsum` is here.

Reply:
Oh nice point!

Reply:
https://github.com/XanaduAI/strawberryfields/pull/377